docs/source/dev/design.rst

```diff
@@ -0,0 +1,99 @@
+.. _development-design:
+
+Design Overview
+===============
+
+This library consists of three conceptual parts:
+
+- The backend, concerned with elementary low-level I/O operations.
+- The I/O-Queue, acting as a communication mechanism and buffer between the other two parts.
+- The user-facing frontend, enforcing the openPMD standard, keeping a logical state of the data, and synchronizing that
+  state with persistent data by scheduling I/O-Tasks.
+
+Backend
+-------
+One of the main goals of this library is to provide a high-level common interface to synchronize persistent data with a
+volatile representation in memory. This includes handling data in any number of supported file formats transparently.
+To enable the user to handle hierarchical, self-describing file formats while disregarding the actual nitty-gritty
```

Comment:
`To <verb>` constructions are not super eloquent. Maybe we can write `In order to enable` or `Consequently/Therefore, enabling users to handle ...`?